exercises/verify.cpp

```diff
@@ -7,12 +7,72 @@ The number 3 is [not] present in the array.
 */
 
 #include <iostream>
+#include <cstdlib>
+#include <ctime>
+#include <limits>
+#define DIM 10
 using namespace std;
 
+template <class Type>
+Type insert_and_control(Type data) // cin control
+{
+  do
+  {
+    cin >> data;
+    if (cin.fail())
+    {
+      cerr << "Insert a number!" << endl;
+      cin.clear();
+      cin.ignore(numeric_limits<streamsize>::max(), '\n');
+    }
+    else
+      break;
+  } while (true);
+  return data;
+}
+
+int *fill_array(int nArr) // fills the array with random numbers between 0 and 15
+{
+  int *array = new int[nArr];
+  for (int i = 0; i < nArr; i++)
+  {
+    array[i] = rand() % 16;
+  }
+  return array;
+}
+
+bool check_array(int array[], int nArr, int nToCheck) // checks whether the given input number is present or not
+{
+  for (int i = 0; i < nArr; i++)
+  {
+    if (array[i] == nToCheck)
+      return true;
+  }
+  return false;
+}
+
+void print_array(int array[], int nArr) // prints the array (duh)
+{
+  cout << "[ ";
+  for (int i = 0; i < nArr; i++)
+  {
+    cout << array[i] << " ";
+  }
+  cout << "]" << endl;
+}
+
 int main()
 {
-  // placeholder
-  int N[10] = [ 3, 4, 5, 1, 2, 3, 4, 9, 13, 0 ];
+  srand(time(0));
+  int *randArray = fill_array(DIM);
+  cout << "Insert number: " << endl;
+  int nToCheck = insert_and_control(nToCheck);
+
+  (!check_array(randArray, DIM, nToCheck)) ? cout << "The number " << nToCheck << " is [not] present in the array." << endl : cout << "The number " << nToCheck << " IS present in the array." << endl;
```

Comment:
mhm troppa ridondanza
```c++
cout << "The number " << nToCheck << " is" << (!check_array(randArray, DIM, nToCheck)) ? " not" : "" << " present in the array." << endl;
```

Reply:
usa la condizione ternaria solo ed esclusivamente dove serve, il resto della frase è la stessa a prescindere dal risultato di check_array()

comunque in C++ viene usato il camelCase, quindi usa checkArray e non lo snake_case

Reply:
> usa la condizione ternaria solo ed esclusivamente dove serve, il resto della frase è la stessa a prescindere dal risultato di check_array()
> 
> comunque in C++ viene usato il camelCase, quindi usa checkArray e non check_array (snake_case)

dovrebbe essere meglio ora (non ho cambiato insert_and_control solo per mantenere coerenza con gli altri codici)

Reply:
ah ok